pkg-manager/resolve-dependencies/src/getCatalogResolver.ts

```diff
@@ -0,0 +1,55 @@
+import { PnpmError } from '@pnpm/error'
+import { parseCatalogProtocol } from '@pnpm/catalogs.protocol-parser'
+import { type Catalogs } from '@pnpm/catalogs.types'
+
+export interface WantedDependency {
+  readonly pref: string
+  readonly alias: string
+}
+
+/**
+ * Dereferences a wanted dependency using the catalog protocol and returns the
+ * configured version.
+ *
+ * Example: catalog:default -> ^1.2.3
+ *
+ * @returns The catalog entry or undefined if the wanted dependency does not use
+ * the catalog protocol.
+ *
+ * @throws {@link PnpmError} if the catalog entry does not exist.
+ */
+export type CatalogResolver = (wantedDependency: WantedDependency) => CatalogResolution | undefined
+
+export interface CatalogResolution {
+  /**
+   * The name of the catalog the resolved specifier was defined in.
+   */
+  readonly catalogName: string
+
+  /**
+   * The specifier that should be used for the wanted dependency.
+   */
+  readonly entrySpecifier: string
+}
+
+export function getCatalogResolver (catalogs: Catalogs): CatalogResolver {
+  return function resolveFromCatalog (wantedDependency: WantedDependency): CatalogResolution | undefined {
```

Comment:
no need to use a nested function. Just use bind.
```suggestion
export function resolveFromCatalog (catalogs: Catalogs, wantedDependency: WantedDependency): CatalogResolution | undefined {
```